vip-filesystem/class.vip-filesystem-stream.php

```diff
@@ -0,0 +1,551 @@
+		}
+
+
+		// Get meta data
+		$meta = stream_get_meta_data( $file );
+		$this->seekable = $meta['seekable'];
+		$this->uri = $meta['uri'];
+
+		$this->file = $file;
+		$this->path = $path;
+
+		return true;
+	}
+
+	/**
+	 * Close a file
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 */
+	public function stream_close() {
+		return $this->close_handler( $this->file );
+	}
+
+	/**
+	 * Check for end of file
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 *
+	 * @return bool
+	 */
+	public function stream_eof() {
+		return feof( $this->file );
+	}
+
+	/**
+	 * Read the contents of the file
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 *
+	 * @param   $count  Number of bytes to read
+	 *
+	 * @return  string  The file contents
+	 */
+	public function stream_read( $count ) {
+		$string = fread( $this->file, $count );
+		if ( false === $string ) {
+			trigger_error( 'Error reading from file: ' . $this->path,
+				E_USER_WARNING );
+			return '';
+		}
+
+		return $string;
+	}
+
+	/**
+	 * Flush to a file
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 *
+	 * @return  bool    True on success. False on failure
+	 */
+	public function stream_flush() {
+		return fflush( $this->file );
+	}
+
+	/**
+	 * Seek a pointer position on a file
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 *
+	 * @param   int   $offset
+	 * @param   int   $whence
+	 *
+	 * @return  bool  True if position was updated, False if not
+	 */
+	public function stream_seek( $offset, $whence ) {
+		if ( ! $this->seekable ) {
+			// File not seekable
+			trigger_error( 'File not seekable: ' . $this->path,
+				E_USER_WARNING );
+			return FALSE;
+		}
+
+		$result = fseek( $this->file, $offset, $whence );
+
+		if ( -1 === $result ) {
+			// Seek failed
+			trigger_error( 'Error seeking on file: ' . $this->path,
+				E_USER_WARNING );
+			return FALSE;
+		}
+
+		return TRUE;
+	}
+
+	/**
+	 * Write to a file
+	 *
+	 * @since   1.0.0
+	 * @accesss public
+	 *
+	 * @param   string      $data   The data to be written
+	 *
+	 * @return  int|bool    Number of bytes written or FALSE on error
+	 */
+	public function stream_write( $data ) {
+		$length = fwrite( $this->file, $data );
+
+		if ( FALSE === $length ) {
+			trigger_error( 'Error writing to file: ' . $this->path,
+				E_USER_WARNING );
+			return FALSE;
+		}
+
+		// Upload to file service
+		$result = $this->client
+			->upload_file( $this->uri, $this->path );
+		if ( is_wp_error( $result ) || $result instanceof \WP_Error ) {
+			trigger_error( $result->get_error_message(), E_USER_WARNING );
+			return FALSE;
+		}
+
+		return $length;
+	}
+
+	/**
+	 * Delete a file
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 * @param   string  $path
+	 *
+	 * @return  bool    True if success. False on failure
+	 */
+	public function unlink( $path ) {
+		$path = $this->trim_path( $path );
+		$result = $this->client->delete_file( $path );
+
+		if ( is_wp_error( $result ) || $result instanceof \WP_Error ) {
```

Comment:
Redundant WP_Error check?

```suggestion
		if ( is_wp_error( $result ) ) {
```